### ISSUE TITLE
Add ability to save and load checker state

### ISSUE THREAD

**Issue by jeffa5** [stateright/stateright#31]
It would be really nice to be able to have the results of a check be saved (including states created/visited etc) so that they could be shared without others having to rerun the check.

Maybe this could lead to the web explorer having an option to upload a file with the check that it can process into an explorable version.

I guess this wouldn't directly mean that the checker can continue from a persisted state but could also enable snapshots while checking is running to enable investigating issues later.

**Comment by jonnadal** [comment]
I like this idea: namely the insight that resuming the check isn't the main value-add. So for instance you could load a snapshot of discoveries and browse them even if you aren't running the checker in the UI, right @jeffa5?

**Comment by jeffa5** [comment]
Definitely! I really like the concept of having a model be explorable and not having to run the checker every time to explore it would be great. I imagine sharing models with other users and maybe running checking on a large machine before analysing locally or offline would be good primary use cases.

Being able to checkpoint the state and resume checking later would also be great but maybe more of an extension.

**Comment by younes-io** [comment]
Support for state persistence could be a game-changer, enabling us to pause/resume checks and share results easily. Leveraging `serde` for serialization seems like a practical path forward. We should also think about a lightweight versioning strategy for the state files to future-proof this feature. Happy to brainstorm on implementation details if needed!

**Comment by jonnadal** [comment]
Hi @younes-io. Sorry for missing your earlier message. If you're open to using [Discord](https://discord.gg/JbxGSVP4A6) we could chat there. In the meantime, my impression is that the code changes wouldn't be too extensive... something along the lines of:

1. Introduce `Checker::snapshot()` that returns a `Serializable` type, and implement for the various checkers. Beyond the runtime state (like pending states and visited fingerprints), I would think this would also encode `CheckerBuilder` parameters and maybe the `Model`.
2. Introduce `CheckerBuilder::from_snapshot(...)` or similar and update the checkers to init runtime state when available rather than calling `Model::init_states()`.
3. Update Explorer to leverage the above for serializing/deserializing to/from disk.
